### PR TITLE
Update the NUClear codebase to the latest version

### PR DESCRIPTION
**Pull request by TrentHouliston**
Updates the NUClear codebase to the latest version to help with some compile errors and improve the stability

### CODE REVIEW

File: .vscode/settings.json
Comment (Bidski) [inline .vscode/settings.json:1]: Looks like you need to add a smudge filter

File: .gitignore [hunk 1]
```diff
@@ -26,3 +26,6 @@ build/Release
 # https://www.npmjs.org/doc/misc/npm-faq.html#should-i-check-my-node_modules-folder-into-git
 node_modules
 build
+
+# # Mac OS X clutter
+# .DS_Store
```
Comment (TrentHouliston) [inline .gitignore:31]:
```suggestion
.DS_Store
```

File: tests/test.js
Comment (JosephusPaye) [inline tests/test.js:1]: Undo the changes in here plz

File: .github/workflows/publish-to-npm.yml [hunk 2]
```diff
@@ -8,20 +8,24 @@ jobs:
...
-          node-version: '16.x'
-          registry-url: 'https://registry.npmjs.org'
-      - run: npm ci
-      # Run the tests up to 5 times if it fails (to avoid the error "Network error when sending
-      # the announce packet: Bad file descriptor" that occurs sometimes on startup)
-      # TODO: investigate and fix the actual issue.
-      - shell: bash
-        run: ./tests/run-with-retries.sh
-      - run: node .github/auto-publish-action.js
+          node-version: "16.x"
```
Comment (JosephusPaye) [inline .github/workflows/publish-to-npm.yml:19]:
```suggestion
          node-version: "18.x"
```

File: .vscode/settings.json
Comment (JosephusPaye) [inline .vscode/settings.json:1]: Delete this

**Review by JosephusPaye — CHANGES_REQUESTED** [review]

File: tests/test.js
Comment (TrentHouliston) [reply tests/test.js:1]: undone

**Review by JosephusPaye — APPROVED** [review]

**Pull request merged**